src/applications/hca/tests/config/serviceInformation.unit.spec.js

```diff
@@ -1,88 +1,89 @@
 import React from 'react';
 import { expect } from 'chai';
 import sinon from 'sinon';
-import ReactTestUtils from 'react-dom/test-utils';
+import { mount } from 'enzyme';
 import moment from 'moment';
 
-import { DefinitionTester, getFormDOM } from '../../../../platform/testing/unit/schemaform-utils.jsx';
+import { DefinitionTester, fillData, fillDate } from '../../../../platform/testing/unit/schemaform-utils.jsx';
 import formConfig from '../../config/form';
 
 describe('Hca serviceInformation', () => {
   const { schema, uiSchema } = formConfig.chapters.militaryService.pages.serviceInformation;
-  const definitions = formConfig.defaultDefinitions;
-  it('should render', () => {
-    const form = ReactTestUtils.renderIntoDocument(
+
+  it('renders military info', () => {
+    const form = mount(
       <DefinitionTester
+        definitions={formConfig.defaultDefinitions}
         schema={schema}
         data={{}}
-        uiSchema={uiSchema}
-        definitions={definitions}/>
+        formData={{}}
+        uiSchema={uiSchema}/>
     );
-    const formDOM = getFormDOM(form);
 
-    expect(formDOM.querySelectorAll('input').length).to.equal(2);
-    expect(formDOM.querySelectorAll('select').length).to.equal(6);
```

Comment:
We now hide discharge type by default, and reveal when discharge date is in the present or past.